google-http-client/src/main/java/com/google/api/client/http/HttpResponse.java

```diff
@@ -356,6 +357,44 @@ public InputStream getContent() throws IOException {
     return content;
   }
 
+  static class ConsumingInputStream extends InputStream {
+    private InputStream inputStream;
+    private boolean closed = false;
+
+    private ConsumingInputStream(InputStream inputStream) {
+      this.inputStream = Preconditions.checkNotNull(inputStream);
+    }
+
+    @Override
+    public int read() throws IOException {
+      return inputStream.read();
+    }
+
+    @Override
+    public int read(byte[] b, int off, int len) throws IOException {
+      return inputStream.read(b, off, len);
+    }
+
+    @Override
+    public void close() throws IOException {
+      if (!closed && inputStream != null) {
+        try {
+          drainInputStream(this);
```

Comment:
Why is it important to drain the input stream and throw away the excess data, instead of simply closing the stream?